### PR TITLE
[OSD-11798] Migrate certman-operator to operator-sdk 1.21

### PR DESCRIPTION
**Pull request by ravitri**
This PR regards the operator-sdk migration work as per [OSD-11798](https://issues.redhat.com/browse/OSD-11798).

Golang - 1.17
Openshift API - 4.11 release
Kubernetes - 1.24
Controller-Runtime - 0.12.1
Operator-SDK - 1.21

### CODE REVIEW

**Comment by ravitri** [comment]
/hold
For development and testing purposes, temporarily pointed to github.com/ravitri/aws-account-operator/api instead of github.com/openshift/aws-account-operator/api. Once PR https://github.com/openshift/aws-account-operator/pull/725 is merged, will point to the openshift repository. Till then, keeping the PR on hold for review.

File: controllers/certificaterequest_controller.go
Comment (yithian) [inline controllers/certificaterequest_controller.go:55]: this doesn't look right to me :laughing:

**Review by yithian — CHANGES_REQUESTED** [review]

File: controllers/certificaterequest/issue_certificate_test.go
Comment (ravitri) [reply controllers/certificaterequest/issue_certificate_test.go:86]: Yup, agreed. Thanks Alex!

File: controllers/certificaterequest_controller.go
Comment (ravitri) [reply controllers/certificaterequest_controller.go:55]: Oh I missed removing this! Thanks to catch it :+1: 

**Background**: What happened here is that when new controller is initialised using `operator-sdk` command, the new file is directly created under `./controllers` directory in the new version. However, as part of the migration I wanted to maintain directory structure so had manually copied controller directories from `./pkg/controller/` in old version. Missed removing this file later on. For the actual controller can refer to this [diff](https://github.com/openshift/certman-operator/pull/225/files#diff-0285b65fbbf63a5eb3c040c52cfac34e69791b73cfc5e3adeb59954b6432a7ad).

File: go.mod
Comment (ravitri) [inline go.mod:21]: **NOTE**: This is temporary till https://github.com/openshift/aws-account-operator/pull/725 is merged. Once it is merged, migrated certman-operator to be tested again with any rebase changes of migrated AAO. If all works, will point to upstream AAO repository then.

File: controllers/certificaterequest/issue_certificate_test.go
Comment (ravitri) [reply controllers/certificaterequest/issue_certificate_test.go]: Thanks Alex! Resolved with [commit](https://github.com/openshift/certman-operator/pull/225/commits/0fd636976046016ef946878a8b2b47e778d23ccc).

File: controllers/certificaterequest/renew_certificate_test.go
Comment (ravitri) [reply controllers/certificaterequest/renew_certificate_test.go]: Thanks Alex! Resolved with [commit](https://github.com/openshift/certman-operator/pull/225/commits/0fd636976046016ef946878a8b2b47e778d23ccc).

**Comment by ravitri** [comment]
Thanks Alex :+1: Have added the fixes based on feedback.

File: controllers/certificaterequest_controller.go
Comment (yithian) [reply controllers/certificaterequest_controller.go]: that makes sense, thanks for clarifying!

File: go.mod
Comment (yithian) [reply go.mod:21]: i don't have any significant concerns once this is resolved

thanks for the big PR!

File: main.go
Comment (mrbarge) [inline main.go:127]: A minor nit, but if `operator-custom-metrics` dependency is bumped up to at least `v0.5.1-0.20220802235640-dc76a1f15ee8` or greater then it should prevent the need for this. (I'm assuming this was added to allow the operator to create the metrics route successfully)

File: go.mod
Comment (ravitri) [reply go.mod]: This has been updated now as the AAO PR is merged.

File: main.go
Comment (ravitri) [reply main.go]: Thanks a lot Matt! Yup, have updated the dependency and tested it as well. Seemed to work fine and certman-operator didn't complain of the route API being unknown.

**Comment by ravitri** [comment]
Based on the testing and comments so far, am ok to unhold this PR (unless anyone has any other concerns)
/unhold

**Comment by mrbarge** [comment]
I've completed testing this operator as part of OSD-12975. A full write-up of testing is available in that ticket, but the tl;dr is that no issues stemming from this PR were identified and the operator is behaving correctly.

**Pull request merged**
